### PR TITLE
Updating and expanding the read me file.

### LINKED ISSUE #16: Add a README.md

### PR DESCRIPTION
**Pull request by kmacdonald-stsci**
Closes #16 

Updating and expanding the README file.

**Checklist**
- [ ] added entry in `CHANGES.rst` (either in `Bug Fixes` or `Changes to API`)
- [ ] updated relevant tests
- [ ] updated relevant documentation
- [ ] updated relevant milestone(s)
- [x] added relevant label(s)

### CODE REVIEW

**Review by hbushouse — COMMENTED** [review]
A lot of this is obviously copied from the jwst README, which is fine (why recreate the wheel?), but some of the descriptions of how and why to do things sound like the environment is being created for standalone use of stcal (like you would for jwst). Is that a use case? Can stcal be installed by itself in an environment and used by itself?  I assume it's more of a support library/package for other higher-level packages like jwst and roman.

**Comment by kmacdonald-stsci** [comment]
> A lot of this is obviously copied from the jwst README, which is fine (why recreate the wheel?), but some of the descriptions of how and why to do things sound like the environment is being created for standalone use of stcal (like you would for jwst). Is that a use case? Can stcal be installed by itself in an environment and used by itself? I assume it's more of a support library/package for other higher-level packages like jwst and roman.

I essentially copied and pasted the JWST README and modified it to be appropriate for STCAL.

Yes.  It can be installed as a stand alone package.  This is how I do all my software updates and developments in STCAL.  When needed I install pipeline packages, such as JWST, to do further testing.  I can make it more clear even though STCAL can be installed and used as a stand alone package it is intended as a dependency/support to pipeline software, such as JWST.

**Review by nden — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -7,3 +7,136 @@
 [![codecov](https://codecov.io/gh/spacetelescope/stcal/branch/main/graph/badge.svg?token=C1LO00W9CZ)](https://codecov.io/gh/spacetelescope/stcal)
 
 STScI Calibration algorithms and tools.
+
+![STScI Logo](docs/_static/stsci_logo.png)
+
+**STCAL requires Python 3.8 or above and a C compiler for dependencies.**
```
Comment (nden) [inline README.md:13]: Astropy is dropping support for Python 3.8 in the next release (coming out some time in may). Let's move to Python 3.9 as a min dependency.

File: README.md [hunk 2]
```diff
@@ -7,3 +7,136 @@
...
+
+**Linux and MacOS platforms are tested and supported.  Windows is not currently supported.**
+
+**If installing on MacOS Mojave 10.14, you must install 
+  into an environment with python 3.8 or 3.9. Installation will fail on python 3.10 due
```
Comment (nden) [inline README.md:18]:
```suggestion
  into an environment with python 3.9. Installation will fail on python 3.10 due
```

File: README.md (hunk 1 above)
Comment (kmacdonald-stsci) [reply README.md]: I made this change.

File: README.md (hunk 2 above)
Comment (kmacdonald-stsci) [reply README.md]: I made this change.

**Review by hbushouse — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -7,3 +7,136 @@
...
+**STCAL requires Python 3.9 or above and a C compiler for dependencies.**
+
+**Linux and MacOS platforms are tested and supported.  Windows is not currently supported.**
+
+**If installing on MacOS Mojave 10.14, you must install 
+  into an environment with python 3.9. Installation will fail on python 3.10 due
+  to lack of a stable build for dependency ``opencv-python``.**
+
+
+`STCAL` is intended to be used as a support package for calibration pipeline
```
Comment (cshanahan1) [inline README.md:22]: I would add a sentence here that mentions Roman and the motivation for this separate package (code common to both JWST + Roman, but could be general use as well), just to make it clearer to someone trying to understand

**Review by cshanahan1 — APPROVED** [review]
Looks fine to me, I left one minor suggestion.

File: README.md (hunk 3 above)
Comment (hbushouse) [reply README.md:22]: I agree with @cshanahan1

Comment (kmacdonald-stsci) [reply README.md:22]: I mentioned Roman to clarify STCAL's usage in multiple pipelines.

**Pull request merged**